api/v1/middleware/auth.js

```diff
@@ -2,29 +2,45 @@ const AuthService = require('../services/AuthService');
 const User = require('../models/User');
 const config = require('../../config');
 const errors = require('../errors');
+const StatusCodeError = require('request-promise/errors').StatusCodeError;
 const roles = require('../utils/roles');
 const _ = require('lodash');
 const logger = require('../../logging');
 
-const AUTH_HEADER = config.auth.header;
-const ADMIN_USER_OVERRIDE_HEADER = 'Admin-User-Override';
+const API_AUTH_HEADER = config.auth.headers.api;
+const GITHUB_AUTH_HEADER = config.auth.headers.github;
+const ADMIN_USER_OVERRIDE_HEADER = config.auth.headers.impersonation;
 
 module.exports = (req, res, next) => {
-  const auth = req.header(AUTH_HEADER);
+  const auth = req.header(API_AUTH_HEADER);
+  const oauth = req.header(GITHUB_AUTH_HEADER);
 
-  if (!auth) {
-    return next();
-  }
+  if(oauth) {
+    return AuthService.getGitHubAccountDetails(oauth)
+    .then((handle) => {
+      req.auth = true;
+      return User.findByGitHubHandle(handle);
+    })
+    .then((user) => {
+      req.user = user;
+      return next();
+    })
+    .catch(StatusCodeError, (error) => {
```

Comment:
If we wrote the service function that's throwing this error, maybe we should catch the `StatusCodeError` and re-throw it with our own internal error (e.g. `UnprocessableRequestError`, like below). Seems kinda cumbersome to have to import another library's error like this, not sure.